server/tests/src/test/java/org/infinispan/server/persistence/PooledConnectionOperations.java

```diff
@@ -112,6 +91,65 @@ public void testPutGetRemove() {
       assertEquals("v2", cache.get("k2"));
       cache.remove("k1");
       assertNull(cache.get("k1"));
+      assertCleanCacheAndStore(cache);
+   }
+
+   @Test
+   public void testPreload() throws Exception {
+      JdbcConfigurationUtil jdbcUtil = new JdbcConfigurationUtil(CacheMode.REPL_SYNC, database, false, true)
+              .setLockingConfigurations();
+      RemoteCache<String, String> cache = SERVER_TEST.hotrod().withServerConfiguration(jdbcUtil.getConfigurationBuilder()).create();
+      cache.put("k1", "v1");
+      cache.put("k2", "v2");
+
+      SERVERS.getServerDriver().stop(0);
+      SERVERS.getServerDriver().restart(0);
+
+      // test preload==true, entries should be immediately in the cache after restart
+      assertEquals(2, cache.size());
+      assertEquals("v1", cache.get("k1"));
+      assertEquals("v2", cache.get("k2"));
+      cache.clear();
+   }
+
+   /*
+    * This should verify that DefaultTwoWayKey2StringMapper on server side can work with ByteArrayKey which
```

Comment:
How is this verifying that sorry?

Reply:
I got this from an old test from jdg-functional-tests, I just removed this test since doesn't makes sense